### PR TITLE
Feature/route refactor

### PR DESCRIPTION
**Pull request by ad2969**
Addresses #87 (separating admin routes and member routes) and refactoring of routes.

The PR branches of #83 because I already included a number of changes to routes in the other PR lol

Mainly, the largest change that this PR includes is a fix of route naming to use [REST convention](https://restfulapi.net/resource-naming/). I thought this would be good for consistency.

### CODE REVIEW

File: src/components/Authentication/LoginRedirect.js [hunk 1]
```diff
@@ -34,9 +58,7 @@ export class LoginRedirect extends Component {
     }
 
     render() {
-        return this.props.user
-            ? <Redirect to="/" />
-            : <CircularProgress />
+        return <div>if this is taking a long time please refresh<CircularProgress /></div>
```
Comment (ianmah) [inline src/components/Authentication/LoginRedirect.js:61]: I think you took the wrong part of the rebase here

File: src/components/Router.js [hunk 2]
```diff
@@ -53,53 +59,65 @@ class Router extends Component {
...
-                path="/new-event"
-                render={() => <EventNew />} />
-              <Route
-                path="/edit-event"
-                render={() => <EventEdit />} />
+                path="/signup"
+                render={() => <UserSignup />} />
               <Route
-                path="/page"
+                path="/event/:id/register"
```
Comment (ianmah) [inline src/components/Router.js:86]: way cleaner 👌

File: src/actions/UserActions.js [hunk 3]
```diff
@@ -3,7 +3,8 @@ import { SET_USER, LOGOUT } from '../constants/Constants';
 export function setUser(userData) {
   const user = {
     email: userData.attributes.email,
-    name: userData.attributes.name
+    name: userData.attributes.name,
+    admin: userData.admin
```
Comment (ianmah) [inline src/actions/UserActions.js:7]: Can't remember why we have `.attributes` for the other fields, does admin not need it?

File: src/components/Authentication/Login.js [hunk 4]
```diff
@@ -37,7 +37,7 @@ function Login(props) {
                 <title>UBC BizTech - Log In or Sign Up</title>
             </Helmet>
             <h1> Login </h1>
-            <Button onClick={() => Auth.federatedSignIn({ provider: 'Google' }).then(() => Auth.currentAuthenticatedUser({ bypassCache: true }))} variant="contained" color="primary">
+            <Button onClick={() => Auth.federatedSignIn({ provider: 'Google' })} variant="contained" color="primary">
```
Comment (ianmah) [inline src/components/Authentication/Login.js:40]: Why was this removed? (tbh can't remember what it's for)

File: src/pages/admin/EventEdit.js [hunk 5]
```diff
@@ -1,12 +1,26 @@
...
 function EventEdit(props) {
+
+    const { id: eventId } = useParams();
+
+    const { setEvent, event, events } = props;
+
+    useEffect(() => {
+        // Reupdate only if event does not exist, and there are events to filter
+        if(!event && events && eventId) {
+            setEvent(events.find(event => event.id === eventId));
```
Comment (ianmah) [inline src/pages/admin/EventEdit.js:20]: I know it was like this before, but do you think it's necessary to store the 'current' event in redux? Maybe we can just use state on the event pages

File: src/actions/UserActions.js (hunk 3 above)
Comment (ad2969) [reply src/actions/UserActions.js:7]: I remember seeing that the `attributes` field is just a follow-up of the format passed from cognito login - it doesn't seem to be necessary for any specific functionality as I can see

File: src/components/Authentication/Login.js (hunk 4 above)
Comment (ad2969) [reply src/components/Authentication/Login.js:40]: Honestly not sure either.. I removed this because it didn't really change anything. `Auth. currentAuthenticatedUser` is already called on every render of `route.js`, so I didn't see why this was needed here.

File: src/components/Authentication/LoginRedirect.js (hunk 1 above)
Comment (ad2969) [reply src/components/Authentication/LoginRedirect.js:61]: Hmm I'm not sure why I put the text here but I did change the redirect logic - instead of relying on a `<Redirect/>` the component will route based on the logic in `useEffect` - through `props.history.push()`

File: src/pages/admin/EventEdit.js (hunk 5 above)
Comment (ad2969) [reply src/pages/admin/EventEdit.js:20]: I thought the same.. As of now, I can't really think of a reason where it would be useful to store individual events except maybe to store a `loaded` boolean state for the page itself.

File: src/components/Authentication/Login.js (hunk 4 above)
Comment (ianmah) [reply src/components/Authentication/Login.js:40]: :seemsgood:

File: src/components/Authentication/LoginRedirect.js (hunk 1 above)
Comment (ianmah) [reply src/components/Authentication/LoginRedirect.js:61]: Ah okay, let's just take out the text and leave just the spinner

File: src/pages/admin/EventEdit.js (hunk 5 above)
Comment (ianmah) [reply src/pages/admin/EventEdit.js:20]: If you would like to remove it, feel free to

File: src/pages/admin/EventEdit.js [hunk 6]
```diff
@@ -11,11 +10,12 @@ import { Helmet } from 'react-helmet';
...
+    const [ event, setEvent ] = useState(null);
 
-    const { setEvent, event, events } = props;
+    const { events } = props;
 
     useEffect(() => {
-        // Reupdate only if event does not exist, and there are events to filter
+        // Get the initial values
         if(!event && events && eventId) {
             setEvent(events.find(event => event.id === eventId));
```
Comment (ianmah) [inline src/pages/admin/EventEdit.js:20]: missed a setEvent here I think

File: src/components/Forbidden.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        textAlign: 'center',
+        height: '100vh',
+    }
+}
+
+const Forbidden = () => {
+
+    return (     
+        <div style={styles.container}>
+            <h1>403: Forbidden</h1>
```
Comment (ianmah) [inline src/components/Forbidden.js:20]: Just had a thought, can we add a helmet/title here

File: src/components/Router.js [hunk 8]
```diff
@@ -55,63 +60,84 @@ class Router extends Component {
...
           <Switch>
+
+            <Route
+              path="/event/:id/register"
+              component={EventRegister} />
+
             <Route
-              path="/page"
-              render={() => <EventRegister />} />
+              path="/signup"
```
Comment (ad2969) [inline src/components/Router.js:130]: changed from `new-member` to `signup` if it makes more sense

File: src/pages/member/NewMemberRegister.js [hunk 9]
```diff
@@ -26,9 +30,19 @@ const useStyles = makeStyles(theme => ({
...
   const classes = useStyles();
-  const { event } = props;
+
+  const history = useHistory();
+  const useQuery = () => { return new URLSearchParams(useLocation().search) };
+  const queries = useQuery();
+  
+  // Destructure query url
+  const initialEmail = queries.get('email');
+  let initialName = queries.get('name');
```
Comment (ianmah) [inline src/pages/member/NewMemberRegister.js:42]: DOn't think we'll have their name before they sign up, only their email

Comment (ad2969) [reply src/pages/member/NewMemberRegister.js:42]: It seems google auth passes a name attribute too (only if available, i think). For facebook auth, will have to check - but even if no name is passed, I made sure there shouldn't be errors popping up.
